src/devices/Pn532/samples/Program.cs

```diff
@@ -164,19 +164,21 @@ void ReadMiFare(Pn532 pn532)
         {
             mifareCard.BlockNumber = block;
             mifareCard.Command = MifareCardCommand.AuthenticationB;
-            var ret = mifareCard.RunMifiCardCommand();
+            var ret = mifareCard.RunMifareCardCommand();
+            // This will reselect the card in case of issue
+            mifareCard.ReselectCard();
             if (ret < 0)
             {
                 // Try another one
                 mifareCard.Command = MifareCardCommand.AuthenticationA;
-                ret = mifareCard.RunMifiCardCommand();
+                ret = mifareCard.RunMifareCardCommand();
```

Comment:
btw why does this have to be called again after running it the first time? Should you call ReselectCard only when ret <0 ?

Reply:
The first one set the command, you may want to adjust the key or the block number, finally the last one run the command itself.

Reply:
Should the command be a argument to RunMifareCardCommand in that case?

Reply:
There are high level integrated commands available as well. this one is convenient in some scenarios. You can see both in the examples btw